Sources/MongoSwift/ChangeStream.swift

```diff
@@ -27,158 +51,130 @@ public struct ResumeToken: Codable, Equatable {
-     *   - `InvalidArgumentError` if the `mongoc_change_stream_t` was created with invalid options.
-     */
     internal init(
-        stealing changeStream: OpaquePointer,
+        stealing changeStreamPtr: OpaquePointer,
         connection: Connection,
         client: MongoClient,
         session: ClientSession?,
         decoder: BSONDecoder,
         options: ChangeStreamOptions?
     ) throws {
-        self.state = .open(changeStream: changeStream, connection: connection, client: client, session: session)
+        let mongocChangeStream = MongocChangeStream(stealing: changeStreamPtr)
+        self.wrappedCursor = try Cursor(
+            mongocCursor: mongocChangeStream,
+            connection: connection,
+            session: session,
+            type: .tailableAwait
+        )
+        self.client = client
         self.decoder = decoder
 
         // TODO: SWIFT-519 - Starting 4.2, update resumeToken to startAfter (if set).
         // startAfter takes precedence over resumeAfter.
         if let resumeAfter = options?.resumeAfter {
             self.resumeToken = resumeAfter
         }
+    }
+
+    deinit {
+        assert(!self.isAlive, "change stream wasn't closed")
+    }
+
+    /// Indicates whether this change stream has the potential to return more data.
+    public var isAlive: Bool {
+        return self.wrappedCursor.isAlive
+    }
 
-        if let err = self.getChangeStreamError() {
-            self.close()
-            throw err
+    /// The `ResumeToken` associated with the most recent event seen by the change stream.
+    public internal(set) var resumeToken: ResumeToken?
+
+    /**
+     * Get the next `T` from this change stream.
+     *
+     * This method will continue polling until a non-empty batch is returned from the server, an error occurs,
+     * or the change stream is killed. Each attempt to retrieve results will wait for a maximum of `maxAwaitTimeMS`
+     * (specified on the `ChangeStreamOptions` passed  to the method that created this change stream) before trying
+     * again.
+     *
+     * - Returns:
+     *   An `EventLoopFuture<T?>` evaluating to the next `T` in this change stream, `nil` if the change stream is
+     *   exhausted, or an error if one occurred. The returned future will not resolve until one of those conditions is
+     *   met, potentially after multiple requests to the server.
+     *
+     *   If the future evaluates to an error, it is likely one of the following:
+     *     - `CommandError` if an error occurs while fetching more results from the server.
+     *     - `LogicError` if this function is called after the change stream has died.
+     *     - `LogicError` if this function is called and the session associated with this change stream is inactive.
+     *     - `DecodingError` if an error occurs decoding the server's response.
+     */
+    public func next() -> EventLoopFuture<T?> {
```

Comment:
so, something users may need to think about if they are using this method is that it is pretty resource-intensive on the thread pool since it continually polls. I'm not sure where the right place to document that consideration is - somewhere on this type? on the `threadPoolSize` parameter? 🤔